lib/trmnl/i18n/synchronization/value_reducer.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module TRMNL
+  module I18n
+    module Synchronization
+      # Reduces a nested structure into key/value pairs with flattened values.
+      ValueReducer = lambda do |initial_value, path = ""|
```

Comment:
This function is very similar to what I had originally written. Doesn't it break the 5-statements-per-method rule?